crates/worker/src/lib.rs

```diff
@@ -331,68 +317,241 @@ impl Worker {
-            );
-            let networking = self.networking.clone();
-            let announce_leader = announce_leader.clone();
-            let mut bifrost = bifrost.clone();
-
-            tc.spawn_child(
-                TaskKind::PartitionProcessor,
-                "partition-processor",
-                Some(processor.partition_id),
-                async move {
-                    let header = Header {
-                        dest: Destination::Processor {
-                            partition_key: *processor.partition_key_range.start(),
-                            dedup: None,
-                        },
-                        source: Source::ControlPlane {},
-                    };
-
-                    let envelope =
-                        Envelope::new(header, Command::AnnounceLeader(announce_leader.clone()));
-                    let payload = Payload::from(envelope.encode_with_bincode()?);
-
-                    // todo: Remove once we have proper leader election
-                    bifrost
-                        .append(LogId::from(processor.partition_id), payload)
-                        .await
-                        .context("failed to write AnnounceLeader record to bifrost")?;
-                    processor.run(networking, bifrost).await
+        // Invoker service
+        tc.spawn_child(TaskKind::SystemService, "invoker", None, self.invoker.run())?;
+
+        let mut latest_epoch_table_version = Version::INVALID;
+        let mut latest_partition_table_version = Version::INVALID;
+        let metadata = metadata();
+
+        let shutdown = cancellation_watcher();
+        tokio::pin!(shutdown);
+
+        let mut partition_processor_manager = PartitionProcessorManager::new(
+            metadata.my_node_id(),
+            self.options,
+            self.rocksdb_storage,
+            self.networking,
+            bifrost,
+            invoker_handle,
+        );
+
+        loop {
+            tokio::select! {
+                _ = &mut shutdown => {
+                    debug!("Initiating shutdown of worker");
+
+                    // This will only shutdown rocksdb writer thread. Everything else will respond to
+                    // the cancellation signal independently.
+                    shutdown_signal.drain().await;
+                    break;
+                },
+                epoch_table_update = metadata.wait_for_version(MetadataKind::EpochTable, latest_epoch_table_version) => {
+                    latest_epoch_table_version = epoch_table_update?.next();
+                    partition_processor_manager.on_epoch_table_update().await?
                 },
-            )?;
+                partition_table_update = metadata.wait_for_version(MetadataKind::PartitionTable, latest_partition_table_version) => {
+                    latest_partition_table_version = partition_table_update?.next();
+                    partition_processor_manager.on_partition_table_update().await?
+                },
```

Comment:
Does a period check of version change enough? Do we need to react on every version update or are we only interested in the latest?

Reply:
I've removed this part until we gain more clarity about the partition processor manager's inputs.